src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -713,15 +716,16 @@ internal override bool Equals(TypeSymbol t2, bool ignoreCustomModifiersAndArrayS
 
             // The checks above are supposed to handle the vast majority of cases.
             // More complicated cases are handled in a special helper to make the common case scenario simple/fast
```

Comment:
Thanks for clarifying. Now I understand: the common case has few locals and smaller stack frame. 
I'll restore the comment.